modules/reranker-cohere/clients/ranker_test.go

```diff
@@ -205,10 +206,14 @@ func (f *testRankHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		if containsDocument(req, "Response 7") {
 			index = 3
 		}
+		f.responseLock.Lock()
 		f.response.Results = f.batchedResults[index]
+		f.responseLock.Unlock()
 	}
 
+	f.responseLock.Lock()
 	outBytes, err := json.Marshal(f.response)
+	f.responseLock.Unlock()
```

Comment:
hmm, I think that this change is not need as at the very top of this method there's a lock for whole method invocation, or is it?
```
f.lock.Lock()
defer f.lock.Unlock()
```

Reply:
but it can stay, it's just a test

Reply:
Ah I had made this change before it hit master. I will rm and just rely on the already existing f.lock